.github/workflows/ci.yml

```diff
@@ -0,0 +1,288 @@
+name: Rust
+on:
+  push:
+    branches:
+      - master
+      - ci # TODO: remove later
+    tags:
+      - '[0-9]+.[0-9]+.[0-9]+'
+  pull_request:
+
+jobs:
+  format:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          submodules: true
+      - name: Setup Rust
+        uses: actions-rs/toolchain@v1
+        with:
+          toolchain: nightly
+          profile: minimal
+          components: rustfmt
+          default: true
+          override: true
+      - name: Cargo cache
+        uses: actions/cache@v2
+        with:
+          path: |
+            ~/.cargo/registry
+            ~/.cargo/git
+          key: ${{ runner.os }}-cargo-rust_stable-${{ hashFiles('**/Cargo.lock') }}
```

Comment:
Shouldn't this be `Cargo.toml`? This is a library after all, this repo contains no .lock file.
(Same goes for the other occurrences.)

Reply:
Of course, maybe the best choice is remove this part of key and use cargo update.

Reply:
There is rust specefic cache action, may be it is a best choice:
https://github.com/marketplace/actions/rust-cache